src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -489,9 +606,10 @@ class npc_mograine : public CreatureScript
         bool SayAshbringer = false;
         int timer = 0;
         int step = 1;
-        bool hasDied;
-        bool heal;
-        bool fakeDeath;
+        bool hasDied = false;
+        bool heal = false;
+        bool fakeDeath = false;
+        Player* playerWhoStartedAshbringer = nullptr;
```

Comment:
private variables hold an _ before the variable names as convention so that it is clear when they are private so `_hasDied`, `_heal` etc